Sources/CombineLongPolling/LongPollingPublisher.swift

```diff
@@ -1,94 +1,115 @@
-//
-//  LongPollingPublisher.swift
-//  CombineExtensions
-//
-//  Created by Luiz Barbosa on 29.10.19.
-//  Copyright © 2020 Lautsprecher Teufel GmbH. All rights reserved.
-//
-
 import Combine
 import Foundation
 
-@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
-extension URLSession {
+/// Represents a Combine publisher that starts a data task that, once finished, will be restarted over and over
+/// again. The data task can terminate due to error or timeout, but this won't be propagated as error to the
+/// LongPollingPublisher, because that would interrupt the main publisher. Instead, this publisher will emit events
+/// of type `Result<(Data, URLResponse), Error>` for every time a `DataTaskPublisher ` emits a successful output
+/// or an error. Then, the `LongPollingPublisher` restarts the `DataTaskPublisher`.
+/// `LongPollingPublisher` will `Never` fail.
+public struct LongPollingPublisher: Publisher {
+    /// The kind of values published by this publisher.
+    public typealias Output = Result<(Data, URLResponse), URLError>
```

Comment:
I understand why you did it like that, but would there be another way to expose "real" errors via the Failure typealias?

Reply:
If you send real failures through the Failure typealias, the subscription will end if any of the requests fail. There's a .retry operator that could recover from that, but then you don't get the error in the end of the pipeline.

Reply:
This is true, and in most cases we probably want to retry on error.

But what happens if we have no connection at all, does it just retry forever, in quick loops? That would be expensive and useless drain of battery.

Reply:
Yes, that's a possibly a problem to be solved by the caller, when receiving error to decide if this is supposed to cancel the subscription. I honestly don't know if this is the best solution, but we must be a little bit resilient at least because some of these requests will fail and we don't want to kill the subscription.
So perhaps a kill switch to kill after N number of consecutive errors, but this is kind of a business logic and not appropriate for this library.
Not sure what is worse. :D